src/sampler.jl

```diff
@@ -0,0 +1,67 @@
+#=
+    `MVNSampler` is used to draw from multivariate normal distribution
+=#
+
+import Base: ==
+
+immutable MVNSampler{TM<:Real,TS<:Real,TQ<:LinAlg.BlasReal}
+    mu::Vector{TM}
+    Sigma::Matrix{TS}
+    Q::Matrix{TQ}
+end
+
+function MVNSampler{TM<:Real,TS<:Real}(mu::Vector{TM}, Sigma::Matrix{TS})
+    TOL1 = 1e-10  # some small value
+    TOL2 = 1e-10  # some small value

```

Comment:
Do numerical experiments by generating singular positive semidefinite matrices and their "slightly non positive semidefinite" perturbations to check the diagonal values of the outputs of their cholfacts and the difference `maxabs(Sigma - Q * Q')`.